library/Zend/Config/Yaml.php

```diff
@@ -289,8 +289,10 @@ protected static function _decodeYaml($currentIndent, &$lines)
     {
         $config   = array();
         $inIndent = false;
-        while (list($n, $line) = each($lines)) {
-            $lineno = $n + 1;
+        while (key($lines) !== null) {
```

Comment:
A simple `foreach` was not possible here because the position of the array pointer is important (both internal to this function (due to the `prev` used below) and external to this function (since `$lines` is passed by reference)).